datafusion/sql/src/planner.rs

```diff
@@ -784,6 +784,9 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
 
                 let filter_expr = self.sql_to_rex(predicate_expr, &join_schema, ctes)?;
 
+                // reduce outer joins
+                let plans = reduce_outer_joins(plans, &filter_expr)?;
```

Comment:
I'm not sure that whether this should be done here, or somewhere else like in optimize phase. Other parts are great :heart: cc @alamb